### PR TITLE
Feat super linter

### PR DESCRIPTION
**Pull request by maryamtahhan**
- Update super linter version to latest
- Fix markdown issues in existing docs
- Fix Typos in existing docs

### CODE REVIEW

File: docs/design/kepler-energy-sources.md [hunk 1]
```diff
@@ -4,80 +4,80 @@
...
 
-The MSRs can be accessed directly on Linux using the msr driver in the kernel. Reading RAPL domain values directly from MSRs requires detecting the CPU model and reading the RAPL energy units before reading the RAPL domain. Once the CPU model is detected, the RAPL domains can be read per package of the CPU by reading the corresponding ’MSR status’ register. 
+The MSRs can be accessed directly on Linux using the msr driver in the kernel. Reading RAPL domain values directly from MSRs requires detecting the CPU model and reading the RAPL energy units before reading the RAPL domain. Once the CPU model is detected, the RAPL domains can be read per package of the CPU by reading the corresponding 'MSR status' register.
```
Comment (maryamtahhan) [inline docs/design/kepler-energy-sources.md:32]:
```suggestion
The MSRs can be accessed directly on Linux using the msr driver in the kernel. Reading RAPL domain values directly from MSRs requires detecting the CPU model and reading the RAPL energy units before reading the RAPL domain. Once the CPU model is detected, the RAPL domains can be read per package of the CPU by reading the corresponding `MSR status` register.
```

File: docs/kepler_model_server/pipeline.md [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
+`pipeline` is composed of three steps, `extract`, `isolate`, and `train`, as shown below. Kepler exports energy-related metrics as Prometheus counter, which provides accumulated number over time. The `extract` step is to convert the counter metrics to gauge metrics, similarly to the Prometheus `rate()` function, giving per-second values. The extract step also clean up the data separately for each `feature group`. The power consumption retrieved from the Prometheus query is the measured power which is composed of the power portion which is varied by the workload called dynamic power and the power portion which is consumed even if in the idling state called idle power. The `isolate` step is to calculate the idle power and isolate the dynamic power consumption of each `energy source`. The `train` step is to apply each `trainer` to create multiple choices of power models based on the preprocessed data.
 
+> We have a roadmap to apply a pipeline to build power models separately for each node/machine type. Find more in [Node Type](#node-type) section.
```
Comment (maryamtahhan) [inline docs/kepler_model_server/pipeline.md:9]:
```suggestion
> **Note**: We have a roadmap to apply a pipeline to build power models separately for each node/machine type. Find more in [Node Type](#node-type) section.
```

File: docs/platform-validation/index.md [hunk 3]
```diff
@@ -60,20 +60,19 @@ We need to introduce an independent and platform agnostic way to collect the nod
 
 Due to the root priviledge limit on the access of the RAPL SYSFS files, we need to use priviledged container to perform the test.
 
-
 For node level power accuracy check, the comparison logic is simple and strightforward, while for pod/container level power accuracy check, the logic is a little bit complicated and need some assumptions.
 
 (1) RAPL energy is node/package level, so we could only use RAPL sampling delta to calculate the power change introduced by application deployment and undeployment.
```
Comment (maryamtahhan) [inline docs/platform-validation/index.md:65]:
```suggestion
1. RAPL energy is node/package level, so we could only use RAPL sampling delta to calculate the power change introduced by application deployment and undeployment.
```

Comment (maryamtahhan) [reply docs/platform-validation/index.md:65]: and the same for the rest of the number list here

File: docs/usage/general_config.md [hunk 4]
```diff
@@ -1,58 +1,59 @@
...
+|Kepler DaemonSet Environment (GPU_METRICS)|gpu|List of performance metrics to enable from gpu source| * (enable all available metrics from gpu source)|
+|***ExportMetric CR*** (single item: default)||||
+|Kepler DaemonSet Environment (PERF_METRICS)|perf|List of performance metrics to export | * (enable all collected performance metrics)|
+|Kepler DaemonSet Environment (EXPORT_NODE_TOTAL_POWER)|node_total_power|Toggle whether to export node total power| true|
+|Kepler DaemonSet Environment (EXPORT_NODE_COMPONENT_POWERS)|node_component_powers|Toggle whether to export node powers by components| true|
+|Kepler DaemonSet Environment (EXPORT_POD_TOTAL_POWER)|pod_total_power|Toggle whether to export pod total power| true|
+|Kepler DaemonSet Environment (EXPORT_POD_COMPONENT_POWERS)|pod_component_powers|Toggle whether to export pod powers by components| true|
+|***EstimatorConfig CR*** (multiple items: node-total-power, node-component-powers, pod-total-power, pod-component-powers)||||
+|Kepler DaemonSet Environment (MODEL_CONFIG.[MODEL_ITEM]_ESTIMATOR)|use-sidecar|Toggle whether to use estimator sidecar for power estimation|false|
```
Comment (maryamtahhan) [inline docs/usage/general_config.md:40]:
```suggestion
|Kepler DaemonSet Environment (MODEL_CONFIG.[`MODEL_ITEM`]_ESTIMATOR)|use-sidecar|Toggle whether to use estimator sidecar for power estimation|false|
```

Here and in the other appearances in the table

**Comment by Shobhij** [comment]
Hi Maryam, Looks like my one wrong github command signed me off against other people commits :P and while trying to solve the I messed up my pull request. Just wondering could you please help me here or point me towards the command which can be helpful?

**Comment by maryamtahhan** [comment]
@rootfs can you please have a quick look. The idea behind this PR is to make existing Markdown docs compliant with the super linter rules.

File: docs/design/metrics.md [hunk 5]
```diff
@@ -206,8 +202,6 @@ resent Intel microarchitecture (consumer-grade/server-grade):
 |      Skylake      |   Y/Y   |     Y/Y    |    Y/**N**   |  Y/Y |
 |     Kaby Lake     |   Y/Y   |     Y/Y    |    Y/**N**   |  Y/Y |
 
-.. _Prometheus: https://prometheus.io
-
-.. _Sysdig: https://sysdig.com/
-
-.. _RAPL: https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/advisory-guidance/running-average-power-limit-energy-reporting.html
+[0]: <https://prometheus.io>
```
Comment (maryamtahhan) [inline docs/design/metrics.md:205]:
```suggestion
[0]:<https://prometheus.io>
```

File: docs/design/metrics.md [hunk 6]
```diff
@@ -42,15 +40,15 @@ All the metrics specific to the Kepler Exporter are prefixed with `kepler_`.
...
+    When [RAPL][3] is not available, kepler can estimate this metric using the model server if the node CPU supports the uncore metric.
 
 - **kepler_container_package_joules_total** (Counter)
     This measures the cumulative energy consumed by the CPU socket, including all cores and uncore components (e.g.
     last-level cache, integrated GPU and memory controller).
     RAPL package energy is typically the PP0 + PP1, but PP1 counter may or may not account for all energy usage
     by uncore components. Therefore, package energy consumption may be higher than core + uncore.
 
-    When `RAPL`_ is not available, kepler might estimate this metric using the model server.
+    When [RAPL][3] is not available, kepler might estimate this metric using the model server.
```
Comment (maryamtahhan) [inline docs/design/metrics.md:51]:
```suggestion
    When [RAPL][2] is not available, kepler might estimate this metric using the model server.
```

**Comment by rootfs** [comment]
@maryamtahhan can you rebase? thanks!

**Review by rootfs — APPROVED** [review]

**Pull request merged**